tests/baselines/reference/genericFunctionAgainstUnresolvableSelfReferencingAwaitedUnion.errors.txt

```diff
@@ -0,0 +1,19 @@
+tests/cases/compiler/genericFunctionAgainstUnresolvableSelfReferencingAwaitedUnion.ts(9,32): error TS2322: Type 'SimpleType' is not assignable to type 'T'.
+  'T' could be instantiated with an arbitrary type which could be unrelated to 'SimpleType'.
+
+
+==== tests/cases/compiler/genericFunctionAgainstUnresolvableSelfReferencingAwaitedUnion.ts (1 errors) ====
+    // repro #49646
+    
+    type EnvFunction = <T>() => T;
+    
+    type SimpleType = string | Promise<SimpleType>;
+    
+    declare const simple: SimpleType;
+    
+    const env: EnvFunction = () => simple;
+                                   ~~~~~~
+!!! error TS2322: Type 'SimpleType' is not assignable to type 'T'.
+!!! error TS2322:   'T' could be instantiated with an arbitrary type which could be unrelated to 'SimpleType'.
+!!! related TS6502 tests/cases/compiler/genericFunctionAgainstUnresolvableSelfReferencingAwaitedUnion.ts:3:20: The expected type comes from the return type of this signature.

```

Comment:
It seems to be close to what 3.8 was reporting:
https://www.typescriptlang.org/play?ts=3.8.3#code/C4TwDgpgBAogdgNwGIFc4GNgEsD2coC8UAPACoB8AFAJSHlSkDcAUM6JFAMpYC2YANhFLhoRAM7AATljgBzKAB8oABUk4eWMRGLc+g4ZHItmAEwjp+AQ0nR0eCVDG8BEAFxdn+kcbtwHERHd4ZDRMXHwiGjpHTwhGIA